hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/Replication.java

```diff
@@ -244,17 +248,22 @@ void addHFileRefsToQueue(TableName tableName, byte[] family, List<Pair<Path, Pat
 
     private final ReplicationSink replicationSink;
     private final ReplicationSourceManager replicationManager;
+    private final MetricsReplicationSourceSource globalMetricsSource;
 
     public ReplicationStatisticsTask(ReplicationSink replicationSink,
-        ReplicationSourceManager replicationManager) {
+        ReplicationSourceManager replicationManager, MetricsReplicationSourceSource globalMetricsSource) {
       this.replicationManager = replicationManager;
       this.replicationSink = replicationSink;
+      this.globalMetricsSource = globalMetricsSource;
     }
 
     @Override
     public void run() {
       printStats(this.replicationManager.getStats());
       printStats(this.replicationSink.getStats());
+
+      // Report how much data we've read off disk which is pending replication, across all sources
+      globalMetricsSource.setWALReaderEditsBufferBytes(replicationManager.getTotalBufferUsed().get());
```

Comment:
this feels odd. Why is it we only do updates for this one global metric here? why is it different then the other things tracked in the global metrics?

Reply:
The rest of the global metrics are tied to some regular "workflow" of replication, e.g. when a new file is queued.

We could piggy-back onto one of those, but I wasn't sure what would be an "appropriate" time in which to report this, so I thought a regular period would be better. Open to suggestions.

Reply:
Pushed this down into each source and it's not too bad.